### DISCUSSION TITLE
Counter restarting with deployments restart

### DISCUSSION THREAD

**Discussion by VictorCallejas** [prometheus/client_python#784] [Q&A] [answered]
Hi,

I have set my grafana dashboard to see some custom metrics of my deployments.

The problem is when the deployment restarts, instead of adding up, the value gets restarted.

![image](https://user-images.githubusercontent.com/32225250/157203033-d39471ee-b913-4711-b0c0-fdae1a1b4808.png)

The query:
`watchdog_observer_files_processed_total{app=~"$watchdog"}`

Underlying this metric is a **Counter**.

Do you know what I can do so I got it cumulative across deployment restarts.

Thanks!</div>

**Comment by csmarchbanks** [comment, marked as answer]
Hello, this is how counters work in Prometheus, and functions such as `rate` and `increase` in Prometheus are designed to handle counter resets. For example, you could use `increase(watchdog_observer_files_processed_total{app=~"$watchdog"}[1h])` to see how many files were processed in the last hour.
